File: ruby/item_usage.rb
# Require all files in the directory
# Dir['./*.rb'].each { |file| require file }

require './item'

mein_item = Item.new('Phone', 100, 5)
puts mein_item.calc_total_price

# There's no way to get all class variables in Ruby

# All instance variables
puts mein_item.instance_variables
# All instance methods
puts mein_item.methods

File: python/item_usage.py
# Fun facts:
# Python is implemented in C, since it is an interpreted language
# Even Ruby & JS are implemented in C/C++, they are interpreted languages too.
# But unlike JS; Ruby & Python have strict typing.
# Ruby & Python are very similar.

from item import Item

mein_item = Item("Phone", 100, 15)

zweites_item = Item("Laptop", 1000, 3)

print(mein_item.calc_total_price())
print(zweites_item.calc_total_price())

print(Item.pay_rate)
print(mein_item.pay_rate)

# Gets all attributes for the class level
print(Item.__dict__)
# Gets all attributes for the instance level
print(mein_item.__dict__)
